src/services/MakePdf.php

```diff
@@ -281,6 +286,26 @@ public function tex2svg(Mpdf $mpdf, string $content): string
         // end
     }
 
+    /**
+     * Get a list of all PDFs that are attached to an entity
+     *
+     * @return array Empty or array of arrays with ['path/to/file', 'real.name']
+     */
+    private function getListOfPdfs(): array
+    {
+        $uploadsArr = $this->Entity->Uploads->readAll();
+        $listOfPdfs = array();
+        if (count($uploadsArr) > 0) {
+            foreach ($uploadsArr as $upload) {
+                $filePath = dirname(__DIR__, 2) . '/uploads/' . $upload['long_name'];
+                if (file_exists($filePath) && preg_match('/(pdf)$/i', Tools::getExt($upload['real_name']))) {
```

Comment:
probably faster to compare strtolower(Tools::getExt) to 'pdf' rather than using regex. Probably doesn't matter though...